lightwood/analysis/nc/icp.py

```diff
@@ -207,17 +209,18 @@ def predict(self, x, significance=None):
                 cal_scores = self.cal_scores[self.condition((x[j, :], c))][::-1]
                 n_cal = cal_scores.size
 
-                idx_left = np.searchsorted(cal_scores, nc, 'left')
-                idx_right = np.searchsorted(cal_scores, nc, 'right')
-                n_gt = n_cal - idx_right
-                n_eq = idx_right - idx_left + 1
-
-                p[j, i] = n_gt / (n_cal + 1)
+                n_eq = 0
+                n_gt = 0
+                for cal_score in cal_scores:
+                    if cal_score == nc:
+                        n_eq += 1
+                    elif nc < cal_score:
+                        n_gt += 1
 
                 if self.smoothing:
```

Comment:
Is this flag typically on? In the prior case, we don't make a random uniform distro UNLESS this flag is true, but here, we make the var then pass it.

Reply:
No it's not, made it so that the dist is created only when it's on, even faster :)

Reply:
See above, this introduces a bug :eyes: